Software/Development/SemesterProject/App/analogAcquisitionManager.c

```diff
@@ -0,0 +1,439 @@
+{
+    readAnalogSignals(&AnalogSignalList, sizeof(AnalogSignalList));
+    calculateFilteredValue(&AnalogSignalList, sizeof(AnalogSignalList));
+}
+
+/*
+ * This takes the address and the size of the struct of interest, and iterates through it to
+ * read the ADC value from the register
+ */
+void readAnalogSignals(void *signal, int size)
+{
+    AnalogSignal *structPointer;
+    AnalogSignal *initialMemoryPosition = signal;
+    AnalogSignal *finalMemoryPosition = initialMemoryPosition + size/sizeof(AnalogSignal);
+
+    for (structPointer = initialMemoryPosition; structPointer < finalMemoryPosition; structPointer++)
+        readADCValue(structPointer);
+}
+
+/*
+ * This takes the address and the size of the struct of interest, and iterates through it to
+ * calculate the filtered voltage measured
+ */
+void calculateFilteredValue(void *signal, int size)
+{
+    AnalogSignal *structPointer;
+    AnalogSignal *initialMemoryPosition = signal;
+    AnalogSignal *finalMemoryPosition = initialMemoryPosition + size/sizeof(AnalogSignal);
+
+    for (structPointer = initialMemoryPosition; structPointer < finalMemoryPosition; structPointer++)
+        filterADCValue(structPointer);
+}
+
+Uint16 getAnalogErrorStatus(void)
+{
+    AnalogSignal *structPointer;
+    AnalogSignal *initialMemoryPosition = &CurrentSignalList.currentMeasA;
+    AnalogSignal *finalMemoryPosition = initialMemoryPosition + sizeof(CurrentSignalList)/sizeof(AnalogSignal);
+
+    Uint16 errorStatus = 0;
+    int i = 0;
+    for (structPointer = initialMemoryPosition; structPointer < finalMemoryPosition; structPointer++)
+    {
+        if (structPointer->filteredValue > structPointer->threshold[0] && structPointer->filteredValue < structPointer->threshold[1])
+            errorStatus |= 1<<i;
+
+        i++;
+    }
+
+    initialMemoryPosition = &AnalogSignalList.voltageMeas24;
+    finalMemoryPosition = initialMemoryPosition + sizeof(AnalogSignalList)/sizeof(AnalogSignal);
+
+    for (structPointer = initialMemoryPosition; structPointer < finalMemoryPosition; structPointer++)
+    {
+        if (structPointer->filteredValue > structPointer->threshold[0] && structPointer->filteredValue < structPointer->threshold[1])
+            errorStatus |= 1<<i;
+
+        i++;
+    }
+
+    return errorStatus;
+}
+
+/*
+ * This interrupt is used to trigger the ADC measurements
+ */
+__interrupt void
+cpu_timer1_isr(void)
+{
+    CpuTimer1.InterruptCount++;
+
+    //
+    // The CPU acknowledges the interrupt
+    //
+    EDIS;
+}
+
+/*
+ * This interrupt is called every time the ADC registers have been updated.
+ */
+__interrupt void adc_isr(void)
+{
+    readHighPrioritySignals();
+    readLowPrioritySignals();
+
+    AdcRegs.ADCINTFLGCLR.bit.ADCINT1 = 1;
+    PieCtrlRegs.PIEACK.all = PIEACK_GROUP1;   // Acknowledge interrupt to PIE
+
+    return;
+}
+
+
+/*
+ * Interface functions to return filtered measurements
+ */
+void getCurrentMeasurements(float *currentMeasurement)

```

Comment:
Add comment before about how to call this function.

Reply:
Done